From d64180e73b3abe2c3784662d2729d0251173332e Mon Sep 17 00:00:00 2001
From: "Lori A. Burns" <lori.burns@gmail.com>
Date: Thu, 3 Apr 2025 00:38:21 -0400
Subject: [PATCH] update math call

---
 tests/test_math.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test_math.py b/tests/test_math.py
index 254c738..46b86ad 100644
--- a/tests/test_math.py
+++ b/tests/test_math.py
@@ -22,7 +22,7 @@ class TestMath(unittest.TestCase):
         # factorial
         rng = np.arange(0, 10, 1)
         for a in rng:
-            assert factorial(a) == np.math.factorial(a)
+            assert factorial(a) == math.factorial(a)
 
     def test_prefactors(self):
         for k in range(3):
-- 
2.40.1

